salt/states/file.py

```diff
@@ -3279,7 +3287,7 @@ def append(name,
             if __salt__['file.search'](
                     name,
                     salt.utils.build_whitespace_split_regex(chunk),
-                    multiline=True):
+                    multiline=True) and ignore_whitespace:
```

Comment:
I'm not sure this is enough to get the change you desire. If I'm not mistaken, if `ignore_whitespace` is False, the text will always be appended, because there's no search for the text while not ignoring whitespace.